src/langrunner.rs

```diff
@@ -205,28 +216,40 @@ impl LangRunnerProcess {
         // Spawn a thread to handle algorithm stderr - we do this here so we shouldn't get get stuck
         // waiting for stuff to be read from stderr when we are loading the algorithm
         let arc_request_id_err = request_id.clone();
+        let arc_stderr_buf = stderr_buf.clone();
         thread::spawn(move || {
             let reader = BufReader::new(stderr);
             for line_result in reader.lines() {
                 let req_id = arc_request_id_err
-                    .read().expect("failed to get read handle on request_id for stdout reading")
+                    .read().expect("failed to get read handle on request_id for stderr reading")
                     .clone().unwrap_or_else(|| "-".to_owned());
                 match line_result {
-                    Ok(line) => info!("{} {} {}", "ALGOERR", req_id, line),
+                    Ok(line) => {
+                        match arc_stderr_buf.lock() {
+                            Ok(mut lines) => {
+                                lines.push_str(&line);
+                                lines.push('\n');
+                            }
+						    Err(err) => error!("{} {} Failed to get lock on stderr buffer: {}", LOG_IDENTIFIER, req_id, err),
+                        }
+                        info!("{} {} {}", "ALGOERR", req_id, line);
+                    },
                     Err(err) => error!("{} {} Failed to read line: {}", LOG_IDENTIFIER, req_id, err),
                 }
             }
         });
 
         let mut reader = BufReader::new(stdout);
+        let arc_stderr_buf = stderr_buf.clone();
```

Comment:
Am I reading this right or do you have 2 variables with the same name in this scope?

Reply:
yep. they're also both references to the same data, as both are `Arc`s cloned from the same underlying variable. Functionally, you can't use the first `arc_stderr_buf` after it was "moved" into a thread, so this just makes an extra `arc_stderr_buf` here.

It might seem strange, but it's not uncommon for rust code to create a new `let-binding` to a previously used variable name that won't be used again; the compiler catches the footguns that make this scary in most languages. But it is equally reasonable to just name them `arc_stderr_buf_1` and `arc_stderr_buf_2`.